tests/php/sync/test_class.jetpack-sync-users.php

```diff
@@ -482,6 +506,103 @@ public function test_delete_user_locale_is_synced() {
 		$this->assertEquals( '', $server_user_local );
 	}
 
+	public function test_delete_user_from_network() {
+		if ( ! is_multisite() ) {
+			$this->markTestSkipped( 'Run it in multi site mode' );
+		}
+
+		$this->server_event_storage->reset();
+
+		wpmu_delete_user( $this->user_id );
+
+		$this->sender->do_sync();
+
+		$event = $this->server_event_storage->get_most_recent_event( 'jetpack_deleted_user' );
+		$this->assertNotEmpty( $event );
+		$this->assertEquals( $this->user_id, $event->args[0] );
+		$this->assertEmpty( $event->args[1] ); //reassign user_id
+		$this->assertTrue( $event->args[2] ); //is network
+
+		$event = $this->server_event_storage->get_most_recent_event( 'jetpack_removed_user_from_blog' );
+		$this->assertNotEmpty( $event );
+
+		$this->assertNull( $this->server_replica_storage->get_user( $this->user_id ) );
+	}
+
+	public function test_remove_user_from_blog_without_reassign() {
+		if ( ! is_multisite() ) {
+			$this->markTestSkipped( 'Run it in multi site mode' );
+		}
+
+		// NOTE this is necessary because WPMU causes certain assumptions about transients
+		// to be wrong, and tests to explode. @see: https://github.com/sheabunge/WordPress/commit/ff4f1bb17095c6af8a0f35ac304f79074f3c3ff6
+		global $wpdb;
+
+		$suppress = $wpdb->suppress_errors();
+		$blog_id = wpmu_create_blog( 'foo.com', '', "My Blog", $this->user_id );
+		$wpdb->suppress_errors( $suppress );
+
+		$original_blog_id = get_current_blog_id();
+		//Put test in same blog that user is removed from
+		switch_to_blog( $blog_id );
+		add_user_to_blog( $this->user_id, $blog_id, 'administrator' );
+
+		$this->server_event_storage->reset();
+
+		remove_user_from_blog( $this->user_id, $blog_id );
+
+		$this->sender->do_sync();
+
+		$event = $this->server_event_storage->get_most_recent_event( 'jetpack_deleted_user' );
+		$this->assertFalse( $event );
+
+		$event = $this->server_event_storage->get_most_recent_event( 'jetpack_removed_user_from_blog' );
+		$this->assertNotEmpty( $event );
+		$this->assertEquals( $this->user_id, $event->args['0'] );
+		$this->assertFalse( $event->args['1'] );
+
+		switch_to_blog($original_blog_id);
```

Comment:
instead of switching back to the original blog you should be able to just use the `restore_current_blog();` function.